build-logic/build-update-utils/src/main/kotlin/gradlebuild/buildutils/tasks/AbstractCheckOrUpdateContributorsInReleaseNotes.kt

```diff
@@ -71,18 +67,31 @@ abstract class AbstractCheckOrUpdateContributorsInReleaseNotes : DefaultTask() {
     @Internal
     protected
     fun getContributorsInReleaseNotes(): Set<GitHubUser> {
-        val releaseNotesText = releaseNotes.asFile.get().readText()
-        if (!contributorSectionRegex.containsMatchIn(releaseNotesText)) {
-            throw IllegalStateException("Can't find the contributors section in the release notes $releaseNotes. You may need to update the regular expression.")
-        }
-        return contributorSectionRegex.find(releaseNotesText)!!.groupValues[2].lines()
+        val (_, contributorLines, _) = parseReleaseNotes()
+        return contributorLines
             .map { it.trim() }
             .filter { it.isNotEmpty() }
             .onEach { if (!contributorLineRegex.containsMatchIn(it)) throw IllegalStateException("Invalid contributor line: $it") }
             .map { GitHubUser(contributorLineRegex.find(it)!!.groupValues[2], contributorLineRegex.find(it)!!.groupValues[1]) }
             .toSet()
     }
 
+    protected
+    fun parseReleaseNotes(): Triple<List<String>, List<String>, List<String>> {
+        val releaseNotesLines: List<String> = releaseNotes.asFile.get().readLines()
+        val contributorSectionBeginIndex = releaseNotesLines.indexOfFirst { it.startsWith("We would like to thank the following community members for their contributions to this release of Gradle:") } + 1
```

Comment:
I wonder how this could be made less brittle. For example removing the `<!- ...` comment after the line would break this already. Maybe it is good enough for now as well, I am not sure.

Reply:
You mean fragile. It was fragile, but currently I think it's robust enough:

1. Find out the contributor begin line by searching `We would like to thank...`
2. Find out the next line which isn't empty or starts with `[` as the contributor end line.

This should be enough for now.

Reply:
If I read the code right, then the start of the contributor section is the line after `We would like to thank...`. Though what I see is that after this line there is a multi line comment,
```
<!-- 
Include only their name, impactful features should be called out separately below.
 [Some person](https://github.com/some-person)
-->
```

And then the lines starting with `[` begin. So with the code we have right now and when the comment is present, the `contributorSectionEndIndex` would also be a line in the comment, since this line isn't empty nor does start with `[`.

I suppose we'd need to increase the  `contributorSectionBeginIndex` until we find the first contributor line starting with a `[`. Or to the first empty line if there are no contributors, yet.

I thought the `+1` was there to skip the comment, though since it is a multiline comment that wouldn't work, right? Also, the multiline comment block contains an example link, aka a line which starts with `[`, so that is also something we'd need to ignore, right? Or maybe detecting the start and the end of the comment would be the way to go.

Reply:
I have a better solution: Move 

```
<!-- 
Include only their name, impactful features should be called out separately below.
 [Some person](https://github.com/some-person)
-->
```

to before `We would like to thank...` in template, so it would be like:

```
<!-- 
Include only their name, impactful features should be called out separately below.
 [Some person](https://github.com/some-person)
-->
We would like to thank the following community members for their contributions to this release of Gradle:
```

Reply:
That is fine with me as well until somebody decides to change the template or the notes again. Maybe we should have a test that the template and the notes are still in good shape? We already check if `We would like to thank the following community...`, so maybe we should also check that the line following that statement is either empty or starts with a `[`?